gammapy/scripts/image_pipe.py

```diff
@@ -2,56 +2,160 @@
+                 energy_band=None, offset_band=None,
+                 data_store=None, obs_table=None, exclusion_mask=None):
+        self.maps = SkyMapCollection()
+
+        self.data_store = data_store
+        if not obs_table:
+            self.obs_table = self.data_store.obs_table
+        else:
+            self.obs_table = obs_table
+        self.energy_band = energy_band
+        self.offset_band = offset_band
+
+        self.empty_image = empty_image
+        self.header = self.empty_image.to_image_hdu().header
+        if exclusion_mask:
+            self.maps['exclusion'] = exclusion_mask
+
+    def make_counts(self, obs_id):
+        """Fill the counts image for the events of one observation.
+
+        Parameters
+        ----------
+        obs_id : int
+            Number of the observation
+
+        """
+        self.maps["counts"] = SkyMap.empty_like(self.empty_image)
+        obs = self.data_store.obs(obs_id=obs_id)
+        events = obs.events.select_energy(self.energy_band)
+        events = events.select_offset(self.offset_band)
+        self.maps["counts"].fill(events=events)
+        self.maps["counts"].data = self.maps["counts"].data.value
         log.info('Making counts image ...')
 
-    def make_exposure(self):
-        log.info('Making exposure image ...')
+    def make_total_counts(self):
+        """Stack the total count from the observation in the 'DataStore'
+
+        """
+        self.maps["total_counts"] = SkyMap.empty_like(self.empty_image)
+        for obs_id in self.obs_table['OBS_ID']:
+            self.make_counts(obs_id)
+            self.maps["total_counts"].data += self.maps["counts"].data
+
+    def make_background(self, obs_id):
+        """Make the background map for one observation from a bkg model
+
+        Parameters
+        ----------
+        obs_id : int
+            Number of the observation
+
+        """
+        self.maps["bkg"] = SkyMap.empty_like(self.empty_image)
+        obs = self.data_store.obs(obs_id=obs_id)
+        table = obs.bkg.acceptance_curve_in_energy_band(energy_band=self.energy_band)
+        center = obs.pointing_radec.galactic
+        bkg_hdu = fill_acceptance_image(self.header, center, table["offset"], table["Acceptance"], self.offset_band[1])
+        livetime = obs.observation_live_time_duration
+        self.maps["bkg"].data = Quantity(bkg_hdu.data, table["Acceptance"].unit) * self.maps[
+            "bkg"].solid_angle() * livetime
+        self.maps["bkg"].data = self.maps["bkg"].data.decompose()
+        self.maps["bkg"].data = self.maps["bkg"].data.value
 
-    def make_background(self):
         log.info('Making background image ...')
 
+    def make_background_normalized_offcounts(self, obs_id):
+        """Normalized the background compare to te events in the counts maps outside the exclusion maps
+
+        Parameters
+        ----------
+        obs_id : int
+            Number of the observation
+
+        """
+        self.make_counts(obs_id)
```

Comment:
Move the scale factor computation in a separate method?
So that the caller can decide whether it should be applied or not?

If you don't think it's worth exposing as a public extra method, you could make it private (but then experts can still compute it without modifying the object if they like), i.e. `_background_norm_factor`.